packages/tools/src/tools/segmentation/strategies/fillRectangle.ts

```diff
@@ -52,13 +57,53 @@ function fillRectangle(
     });
   });
 
-  const boundsIJK = getBoundingBoxAroundShape(
+  const boundsIJK = getBoundingBoxAroundShapeIJK(
     rectangleCornersIJK,
     segmentationImageData.getDimensions()
   );
 
-  // Since always all points inside the boundsIJK is inside the rectangle...
-  const pointInRectangle = () => true;
+  const isStackViewport = viewport instanceof StackViewport;
+
+  // Are we working with 2D rectangle in axis aligned viewport view or not
+  const isAligned =
+    isStackViewport || isAxisAlignedRectangle(rectangleCornersIJK);
+
+  const direction = segmentationImageData.getDirection();
+  const spacing = segmentationImageData.getSpacing();
+  const { viewPlaneNormal } = viewport.getCamera();
+
+  // In case that we are working on oblique, our EPS is really the spacing in the
+  // normal direction, since we can't really test each voxel against a 2D rectangle
+  // we need some tolerance in the normal direction.
+  const EPS = csUtils.getSpacingInNormalDirection(
```

Comment:
Half this spacing would be ok, since the center point +/- 1/2 the normal direction spacing should include all pixels, whereas if you are +/- 1 full pixel, then you could end up including two pixels when you are trying to fill a 1 pixel thick plane.

Reply:
Can you get into a situation where it includes two pixels in adjacent planes?  If so, then you might want to try +/- 3/4
I think the +/- 1 is too large, but obviously the +/- 1/2 is just a tiny bit too small.  I suspect hitting particular bounds exactly just over the edge.  You might try +/- 1/2 + epsilon